src/main/java/org/testng/internal/reflect/ReflectionRecipes.java

```diff
@@ -303,6 +303,9 @@ public static boolean lenientMatch(final Class<?>[] classes, final Object[] args
       for (final Parameter parameter : parameters) {
         boolean omit = false;
         for (final InjectableParameter injectableParameter : filters) {
+          if (injectableParameter == null) {
```

Comment:
As it is internal code, could we enforce that we don't pass null values in the set?

Reply:
I realized we don't need this null check at all, because we are iterating a `Set` and `Set` doesn't allow `null` values. So fixed this by removing this check.